plugins/modules/gitlab_merge_request.py

```diff
@@ -0,0 +1,415 @@
+  community.general.gitlab_merge_request:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: "group1/project1"
+    source_branch: branch1
+    target_branch: branch2
+    title: "Ansible demo MR"
+    state_filter: "opened"
+    state: absent
+'''
+
+RETURN = r'''
+msg:
+  description: Success or failure message.
+  returned: always
+  type: str
+  sample: "Success"
+
+mr:
+  description: API object.
+  returned: success
+  type: dict
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.api import basic_auth_argument_spec
+from ansible.module_utils.common.text.converters import to_native, to_text
+
+from ansible_collections.community.general.plugins.module_utils.version import LooseVersion
+from ansible_collections.community.general.plugins.module_utils.gitlab import (
+    auth_argument_spec, gitlab_authentication, gitlab, ensure_gitlab_package
+)
+
+
+class GitlabMergeRequest(object):
+
+    def __init__(self, module, project, gitlab_instance):
+        self.repo = gitlab_instance
+        self._module = module
+        self.project = self.get_project(project)
+
+    '''
+    @param project Name of the project
+    '''
+    def get_project(self, project):
+        try:
+            return self.repo.projects.get(project)
+        except gitlab.exceptions.GitlabGetError as e:
+            self._module.fail_json(msg="Failed to get the project: %s" % to_native(e))
+
+    '''
+    @param branch Name of the branch
+    '''
+    def get_branch(self, branch):
+        try:
+            return self.project.branches.get(branch)
+        except gitlab.exceptions.GitlabGetError as e:
+            self._module.fail_json(msg="Failed to get the branch: %s" % to_native(e))
+
+    '''
+    @param title Title of the Merge Request
+    @param source_branch Merge Request's source branch
+    @param target_branch Merge Request's target branch
+    @param state_filter Merge Request's state to filter on
+    '''
+    def get_mr(self, title, source_branch, target_branch, state_filter):
+        mrs = []
+        try:
+            mrs = self.project.mergerequests.list(search=title, source_branch=source_branch, target_branch=target_branch, state=state_filter)
+        except gitlab.exceptions.GitlabGetError as e:
+            self._module.fail_json(msg="Failed to list the Merge Request: %s" % to_native(e))
+
+        if len(mrs) > 1:
+            self._module.fail_json(msg="Multiple Merge Requests matched search criteria.")
+        if len(mrs) == 1:
+            try:
+                return self.project.mergerequests.get(id=mrs[0].iid)
+            except gitlab.exceptions.GitlabGetError as e:
+                self._module.fail_json(msg="Failed to get the Merge Request: %s" % to_native(e))
+
+    '''
+    @param username Name of the user
+    '''
+    def get_user(self, username):
+        try:
+            users = self.project.users.list(search=username, all=True)
```

Comment:
I think it may be better to use the `username` parameter instead of `search`, to make GitLab browsing only usernames and not email addresses.